sgt-back-end/index.js

```diff
@@ -0,0 +1,147 @@
+const pg = require('pg');
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const express = require('express');
+const app = express();
+app.use(express.json());
+
+app.get('/api/grades/:gradeId', (req, res, next) => {
+  const gradeId = Number(req.params.gradeId);
+  if (!Number.isInteger(gradeId) || gradeId <= 0) {
+    res.status(400).json({
+      error: '"gradeId" must be a positive integer'
+    });
+    return;
+  }
+  const sql = `
+    select "gradeId",
+           "name",
+           "course",
+           "score",
+           "createdAt"
+      from "grades"
+     where "gradeId" = $1
+  `;
+  const params = [gradeId];
+
+  db.query(sql, params)
+    .then(result => {
+      const grade = result.rows[0];
+      if (!grade) {
+        res.status(404).json({
+          error: `Cannot find grade with "gradeId" ${gradeId}`
+        });
+      } else {
+        res.json(grade);
+      }
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
+});
+
+app.get('/api/grades', (req, res, next) => {
+  const sql = `
+    SELECT * FROM "grades"
+    `;
+  db.query(sql).then(result => {
+    const grades = result.rows;
+    return res.status(200).json(grades);
+  }).catch(err => {
+    console.error(err);
+    return res.status(500).json({ error: 'An unexpected error occured' });
+  });
+});
+
+app.post('/api/grades', (req, res, next) => {
+  if (!req.body.name || !req.body.course || !req.body.score) {
+    return res.status(400).json({ Error: 'Missing name, course, or score.' });
```

Comment:
Zero is a valid score

Reply:
To be clear, a `score` of `0` _should_ be treated as valid. Right now, it's triggering a `400 Bad Request`.